plugins/gatsby-source-figma/gatsby-node.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+const fetch = require('node-fetch');
+const { createRemoteFileNode } = require('gatsby-source-filesystem');
+
+const NODE_TYPE = 'FIGMA_ASSET';
+
+// Flattens a tree into a list and associates nodes using a parent-child relationship
+const createNodeList = root => {
+  const nodes = [root];
+  const nodeList = [];
+
+  while (nodes.length) {
+    let node = nodes[0];
+
+    if (node.children) {
+      const children = node.children.map(child => ({ ...child, parent: node }));
+
+      nodes.push(...children);
+    }
+
+    node = nodes.shift();
+    nodeList.push(node);
+  }
+
+  return nodeList;
+};
+
+exports.sourceNodes = async (
+  { actions: { createNode }, createContentDigest, cache, reporter },
+  configOptions
+) => {
+  const baseUrl = 'https://api.figma.com/v1';
+  const headers = {
+    'Content-Type': 'application/json',
+    'X-FIGMA-TOKEN': configOptions.figmaApiToken
+  };
+
+  try {
+    const response = await fetch(`${baseUrl}/files/${configOptions.fileId}`, { headers });
+    const { document } = await response.json();
+    const nodes = createNodeList(document).filter(node => node.type === 'FRAME');
```

Comment:
The alternative to generating a list of nodes would be to directly traverse the tree (`document`). The drawback of directly traversing the tree is that image URLs would need to be fetched one at a time, resulting in `N` network requests where `N` is the number of Figma frames. 

Instead, I went with generating a list of nodes, which allows for fetching all the image URLs in a single request.